File: clients/javascript/model/text-alignment.js
class TextAlignment {
    constructor (discriminant) {
        this.discriminant = discriminant;
    }

    static async readFrom (stream) {
        switch (await stream.readInt()) {
        case 0:
            return new TextAlignment(TextAlignment.LEFT);
        case 1:
            return new TextAlignment(TextAlignment.CENTER);
        case 2:
            return new TextAlignment(TextAlignment.RIGHT);
        default:
            throw new Error('Unexpected discriminant value');
        }
    }
}

TextAlignment.LEFT = 0;
TextAlignment.CENTER = 1;
TextAlignment.RIGHT = 2;

module.exports.TextAlignment = TextAlignment;

File: clients/scala/src/main/scala/model/TextAlignment.scala
package model

import util.StreamUtil

sealed abstract class TextAlignment (val discriminant: Int) {
    def writeTo(stream: java.io.OutputStream) {
        StreamUtil.writeInt(stream, discriminant)
    }
}

object TextAlignment {
    case object LEFT extends TextAlignment(0)
    case object CENTER extends TextAlignment(1)
    case object RIGHT extends TextAlignment(2)
    def readFrom(stream: java.io.InputStream): TextAlignment = StreamUtil.readInt(stream) match {
        case 0 => LEFT
        case 1 => CENTER
        case 2 => RIGHT
        case _ => throw new java.io.IOException("Unexpected discriminant value")
    }
}
